File: storage/models/protocol_performance_entry.go
package models

type ProtocolPerformanceEntry struct {
	Model
	TestProtocolID uint    `json:"testProtocolId"`
	Checkpoint     string  `json:"checkpoint"`
	FPS            float64 `sql:"type:decimal(10,2);" json:"fps"`
	MEM            float64 `sql:"type:decimal(10,2);" json:"mem"`
	CPU            float64 `sql:"type:decimal(10,2);" json:"cpu"`
	Other          string  `json:"other"`
	Runtime        float64 `sql:"type:decimal(10,2);" json:"runtime"`
	VertexCount    float64 `sql:"type:decimal(10,2);" json:"vertexCount"`
	Triangles      float64 `sql:"type:decimal(10,2);" json:"triangles"`
}

File: frontend/src/types/protocol.performance.entry.ts

export default interface IProtocolPerformanceEntryData {
    id?: number | null,
    testProtocolId: number,
    checkpoint: string,
    fps: number,
    mem: number,
    cpu: number,
    vertexCount: number,
    triangles: number,
    other: string,
    runtime: number,
    executionTime: number,
    createdAt: Date,
    updatedAt: Date,
    deletedAt: Date,
}
